mlir/CMakeLists.txt

```diff
@@ -78,6 +78,72 @@ endif()
 add_subdirectory(include)
 add_subdirectory(lib)
 add_subdirectory(tools)
+
+##############################################################################
+# The following subsection of code                                           #
+# was taken from the CIRCT project: https://github.com/llvm/circt            #
+# Small alteration were made for Catalyst                                    #
+# The CIRCT project has a the following license:                             #
```

Comment:
```suggestion
# The CIRCT project has the following license:                             #
```

Reply:
Alternatively, we recently had to copy and modify some code from JAX, and the copyright notice we settled on is the one used here: https://github.com/PennyLaneAI/catalyst/blob/main/frontend/catalyst/utils/jax_cpu_lapack_kernels/lapack_kernels.hpp#L15-L19, so we could use the same format here.

Reply:
Mmm... I'm having some trouble knowing what to write `// Copyright 2021 ${HERE}.` would it just be the LLVM authors? The LLVM/CIRCT authors?

Reply:
@josh146

Reply:
I would put `// Copyright 2024 The LLVM Project.`

Reply:
@joeycarter and @josh146 I feel like the current copyright notice is better as it reflects the slight separation between CIRCT and LLVM. I am not a lawyer though. Happy to accept suggestions for change here or comments.

Reply:
I'm no lawyer either but I think it's good as is! The notice you have should satisfy the license and copyright notice requirements of the Apache License 2.0.